### PR TITLE
Added support for other forks Banano, NOS, Nollar

### PR DESCRIPTION
**Pull request by tmackness**
To be honest could probably do this a little better but I didn't want to rearrange your params too much.

Didn't get around to adding unit testing.

### CODE REVIEW

**Comment by marvinroger** [comment]
Hi @tmackness 
Thanks for your PR. I am not really in favor of supporting forks (I am not against Banano/NOS/Nollar, I am just not familiar with them).

If it's only a matter of changing the prefixes, I am fine with it. But, will everything (other than the prefix) remain "compatible" with Nano?

**Comment by tmackness** [comment]
Hi,

Yeah it’s only the prefixes that is needed to support them that’s it. I’m working with Nollar so would be good to support it. 

Alternatively, you could just have a user defined prefix but I don’t feel that is the best way Incase a user misspells the prefix.

File: packages/nanocurrency/src/keys.ts [hunk 1]
```diff
@@ -87,6 +87,12 @@ export function derivePublicKey(secretKeyOrAddress: string): string {
 export interface DeriveAddressParams {
   /** Whether to use nano_ instead of xrb_ */
   useNanoPrefix?: boolean
+  /** Whether to use NOS prefix nos_ instead of xrb_ or nano_ */
+  nos?: boolean
```
Comment (marvinroger) [inline packages/nanocurrency/src/keys.ts]: I'd be more comfortable with a `useFork?: Fork` param, with `Fork` enum.

Comment (tmackness) [reply packages/nanocurrency/src/keys.ts]: Yeah that is prob much easier to support them without having to worry about testing etc. 

Thanks

**Comment by tmackness** [comment]
Also how would you feel about making nano_ the primary now that Nano has done this too. Maybe can do a major update to prevent breaking current applications.

**Comment by marvinroger** [comment]
Okay, I reviewed the changes needed to implement support for forks, and it's much more involved than that:

* The library checks for `xrb_|nano_` all over the places
* The `deriveAddress` and `createBlock` supports only `nano_` and `xrb_` prefixes. We have to make sure that the input matches the output format for `createBlock`

There are a couple of other things to think about, which make supporting forks a non-trivial task.

Thinking about it, the library is named `nanocurrency-js`. I do support Nano and I don't want to introduce any confusions for users, when, if another prefix is needed, all you have to do is `nanoAddress.replace('nano_', 'ban_')`.

Thanks for your PR anyway, @tmackness 😉

**Comment by tmackness** [comment]
No worries, yeah I did think of that initially but thought it might be cool to introduce it as well. Yeah I actually hadn't thought that far ahead about the other functions to check if they used the prefix.

**Comment by thelamer** [comment]
Sorry to ping on a necro pull @marvinroger but couldn't a global be set by the user optionally using like `NanoCurrency.setPrefix('ban_')` in their codebase and then just modify parse.ts and keys.ts to optionally use this global var first if set? 
This would ensure no functions need to change to accept new prefix values and you would just use the prefix if set across all of the logic including block generation and address checking. 

Unless I am missing something this should support any Nano forks out of the box with address prefix changes.

**Pull request closed without merging**